agent/agentssh/exec_other.go

```diff
@@ -0,0 +1,24 @@
+//go:build !windows
+
+package agentssh
+
+import (
+	"context"
+	"os/exec"
+	"syscall"
+
+	"cdr.dev/slog"
+)
+
+func cmdSysProcAttr() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		Setsid: true,
+	}
+}
+
+func cmdCancel(ctx context.Context, logger slog.Logger, cmd *exec.Cmd) func() error {
+	return func() error {
+		logger.Debug(ctx, "cmdCancel: sending SIGHUP to process and children", slog.F("pid", cmd.Process.Pid))
```

Comment:
📋🍝 suggestion: check that `cmd.Process != nil` or capture the pid outside of the returned function.

Reply:
Checking nil isn't useful here as it's used in `cmd.Cancel = cmdCancel(...)` meaning this will only get triggered if a cmd is started. Capturing the pid outside also happens _before_ the cmd is started to `cmd.Process` will be nil at that point.